### PR TITLE
Duplo 11472

### PR DESCRIPTION
**Pull request by ddiaz-duplo**
## Change description

Add examples for Infrastructure settings and lambda permissions

## Related issues

https://app.clickup.com/t/8655600/DUPLO-11171
https://app.clickup.com/t/8655600/DUPLO-11472

### Development

- [x] Pull requests may not be submitted to the *master* branch (use *develop* instead) - or they will be closed.
- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-11472 TF: Doc:Example for infrastructure setting](https://app.clickup.com/t/8655600/DUPLO-11472)

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-11171 TF: Add example usage for `duplocloud_aws_lambda_permission` resource](https://app.clickup.com/t/8655600/DUPLO-11171)

File: examples/resources/duplocloud_aws_lambda_permissions/resource.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+resource "aws_lambda_permission" "permission" {
```
Comment (joek-duplo) [inline examples/resources/duplocloud_aws_lambda_permissions/resource.tf:1]: this is not an example of duplo lambda permissions, it is an example of AWS's lambda permission. what are we trying to accomplish here?

File: examples/resources/duplocloud_infrastructure_setting/resource.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
+resource "duplocloud_infrastructure_setting" "settings" {
```
Comment (joek-duplo) [inline examples/resources/duplocloud_infrastructure_setting/resource.tf:1]: this example is way too complicated for most of our users.  please remove the:
 - dynamic block
 - lifecycle block
 - typed variable
 
instead, just show how to use this TF resource itself

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: examples/resources/duplocloud_aws_lambda_permissions/resource.tf (hunk 1 above)
Comment (ddiaz-duplo) [reply examples/resources/duplocloud_aws_lambda_permissions/resource.tf:1]: Good catch! thats why `make doc` wasn't working.

File: examples/resources/duplocloud_infrastructure_setting/resource.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
```
Comment (joek-duplo) [inline examples/resources/duplocloud_infrastructure_setting/resource.tf:1]: docs for this resource are now useless, right?  we need proper docs

File: examples/resources/duplocloud_infrastructure_setting/resource.tf [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+  infra_name            = duplocloud_infrastructure.myinfra
+  EnableSecretCsiDriver = "true"
```
Comment (joek-duplo) [inline examples/resources/duplocloud_infrastructure_setting/resource.tf:3]: this is incorrect

File: examples/resources/duplocloud_infrastructure_setting/resource.tf [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+  EnableAWSEfsVolumes   = "true"
```
Comment (joek-duplo) [inline examples/resources/duplocloud_infrastructure_setting/resource.tf:4]: this is incorrect

File: examples/resources/duplocloud_aws_lambda_permissions/import.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# Example: Importing an existing AWS lambda function permissions
```
Comment (joek-duplo) [inline examples/resources/duplocloud_aws_lambda_permissions/import.sh:1]: this file is ineffective

File: examples/resources/duplocloud_aws_lambda_permissions/resource.tf [hunk 7]
```diff
@@ -0,0 +1,30 @@
+resource "duplocloud_aws_lambda_permission" "permission" {
```
Comment (joek-duplo) [inline examples/resources/duplocloud_aws_lambda_permissions/resource.tf:1]: this file is ineffective

File: examples/resources/duplocloud_infrastructure_setting/import.sh [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# Example: Importing an existing infrastructure setting
+#  - *NAME* is the infrastructure name
+#
+terraform import duplocloud_infrastructure_setting.settings adminproxy/GetInfrastructureConfig/*NAME*
```
Comment (joek-duplo) [inline examples/resources/duplocloud_infrastructure_setting/import.sh:4]: this is incorrect

File: duplosdk/aws_lambda_function.go [hunk 9]
```diff
@@ -256,7 +256,7 @@ func (c *Client) LambdaPermissionDelete(tenantID, functionName, statementId stri
 }
 
 func (c *Client) LambdaPermissionGet(tenantID string, functionName string) (*[]DuploLambdaPermissionStatement, ClientError) {
-	rp := []DuploLambdaPermissionStatement{}
+	var rp []DuploLambdaPermissionStatement
```
Comment (joek-duplo) [inline duplosdk/aws_lambda_function.go:259]: revert this change

**Review by joek-duplo — CHANGES_REQUESTED** [review]

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**